doc/wol/Wake-on-LAN-HLD.md

```diff
@@ -88,19 +92,56 @@ Offset   |0     |1     |2     |3     |4     |5     |
          +-----------------------------------------+
 ```
 
+### Magic Pattern in UDP Payload
+**Magic Packet** is a Ethernet frame with structure:
+
+* **IP Header**:
+  * **Destination IP**: user provided target IP address, cloud be IPv4 address or IPv6 address, default is `255.255.255.255`.
+* **UDP Header**:
+  * **Destination Port**: user provided target port, default is 9.
+* **UDP Payload**:
+  * Six bytes of all `0xff`. [6 bytes]
+  * Sixteen repetitions of the target device's MAC address. [96 bytes]
+  * (Optional) A four or six byte password. [4 or 6 bytes]
+
+```
+Byte     |
+Offset   |0     |1     |2     |3     |4     |5     |
+---------+------+------+------+------+------+------+
+           \      \      \      \      \      \     
+         +------+------+------+------+------+------+
+         |.. Destination IP(4 bytes or 16 bytes) ..|
+         +------+------+------+------+------+------+      
+           \      \      \      \      \      \      
+         +------+------+---------------------------+
+         |  ... Destination Port(2 bytes) ...      |   
+         +------+------+---------------------------+
+           \      \      \      \      \      \      
+         +------+------+------+------+------+------+------------------------------
+       N | 0xFF | 0xFF | 0xFF | 0xFF | 0xFF | 0xFF |
+         +------+------+------+------+------+------+           UDP
+     N+6 | Target MAC (repeat 16 times, 96 bytes)  |           PAYLOAD
+         +-----------------------------------------+
+   N+102 |    Password (optional, 4 or 6 bytes)    |
+         +-----------------------------------------+
+```
+
 ## CLI Design
 
 The `wol` command is used to send magic packet to target device.
 
 ### Usage
 
 ```
-wol <interface> <target_mac> [-b] [-p password] [-c count] [-i interval]
+wol <interface> <target_mac> [-b] [-u] [-a ip-address] [-t udp-port] [-p password] [-c count] [-i interval]
 ```
 
 - `interface`: SONiC interface name.
 - `target_mac`: a list of target devices' MAC address, separated by comma.
 - `-b`: Use broadcast MAC address instead of target device's MAC address as **Destination MAC Address in Ethernet Frame Header**.
+- `-u`: Let device send magic pattern in udp payload, when this flag was used, -b won't work and destination mac is set by device's networking stack.
```

Comment:
This description is not consistent with your UT update. Which is your expected behavior?
1. User can use `-b` and `-u` together, but CLI will ignore `-b` when `-u` exists.
2. CLI return with `ParamValidateError` when `-b` and `-u` are used together.
Please also mention this restriction in `-b` param.

Reply:
It's expected that CLI will return with ParamValidateError when -b and -u are used together.